lib/active_model/serializer.rb

```diff
@@ -108,20 +110,29 @@ def self.digest_caller_file(caller_line)
       Digest::MD5.hexdigest(serializer_file_contents)
     end
 
-    def self.get_serializer_for(klass)
+    def self.get_serializer_for(klass, parent_serializer)
```

Comment:
can you add some yardoc to various parts of this method? explaining what each section does, and what scenarios will be applicable to each block of code?